api/src/test/java/jakarta/data/displaynamegeneration/test/ReplaceCamelCaseAndUnderscoreAndNumberTest.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2022,2023 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package jakarta.data.displaynamegeneration.test;
+
+import jakarta.data.displaynamegeneration.ReplaceCamelCaseAndUnderscoreAndNumber;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayNameGeneration;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.SoftAssertions.assertSoftly;
+
+@DisplayNameGeneration(ReplaceCamelCaseAndUnderscoreAndNumber.class)
+class ReplaceCamelCaseAndUnderscoreAndNumberTest {
+    private ReplaceCamelCaseAndUnderscoreAndNumber replaceCamelCaseAndUnderscoreAndNumber;
+
+    @BeforeEach
+    void setUp() {
+        replaceCamelCaseAndUnderscoreAndNumber = new ReplaceCamelCaseAndUnderscoreAndNumber();
+    }
```

Comment:
I don't think you need to create a new instance of the generator each time.

Reply:
Exactly,  I thought the same,  I will create a static final INSTANCE and use it instead, something similar to the inner class Simple